src/mappings/common.ts

```diff
@@ -17,10 +17,20 @@ export function isBatch(call: CallBase<AnyTuple>) : boolean {
     return call.section == "utility" && batchCalls.includes(call.method)
 }
 
+export function isProxy(call: CallBase<AnyTuple>) : boolean {
+    return call.section == "proxy" && call.method == "proxy"
+}
+
 export function callsFromBatch(batchCall: CallBase<AnyTuple>) : CallBase<AnyTuple>[] {
     return batchCall.args[0] as Vec<CallBase<AnyTuple>>
 }
 
+export function callsFromProxy(proxyCall: CallBase<AnyTuple>) : CallBase<AnyTuple>[] {
+    let args = proxyCall.args[2] as CallBase<AnyTuple>
+    let calls = args.args[0] as Vec<CallBase<AnyTuple>>
```

Comment:
What if it is not a batch? I suggest to extract a call from proxy and assign it to the causeCall. Then repeat your main procedure determining if it is a proxy, batch or regular call.